src/goto-symex/goto_symex_state.cpp

```diff
@@ -306,6 +188,41 @@ static void assert_l2_renaming(const exprt &expr)
   #endif
 }
 
+class goto_symex_is_constantt : public is_constantt
+{
+protected:
+  bool is_constant(const exprt &expr) const override
+  {
+    if(expr.id()==ID_mult)
+    {
+      // propagate stuff with sizeof in it
+      forall_operands(it, expr)
+      {
+        if(it->find(ID_C_c_sizeof_type).is_not_nil())
+          return true;
+        else if(!is_constant(*it))
+          return false;
+      }
+
+      return true;
+    }
+    else if(expr.id()==ID_with)
+    {
+      // this is bad
```

Comment:
This is probably not your code, but it would be good to give an idea of 'what didn't work here' in place of a commented piece of code.

Reply:
Indeed, let me try to dig this up.

Reply:
(But it might take a while to do so.)

Reply:
Throughout the back-end there are hints that the array theory used to do Strange Things.  At one point I (tried to) patch a rewrite that converted with to ITE.

Reply:
#3095 for further discussion.